### PR TITLE
Verify ledger hash after migration

### PR DESCRIPTION
**Pull request by ghost-not-in-the-shell**
Explain your changes:
This PR adds ledger_hash check to `berkeley_migration_verifier`, it also change the input to `fork_config_file` from the output of pre-fork state export to the new genesis config file of the new chain. 

Explain how you tested your changes:
*

Checklist:

- [ ] Dependency versions are unchanged
  - Notify Velocity team if dependencies must change in CI
- [ ] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them

* Closes #0000

### CODE REVIEW

**Review by dkijania — COMMENTED** [review]
I would also increase verbosity when ledger hash is different than expected, by guiding user how to detect what is a change so either link to your script which is comparing ledgers with jq or maybe we can skip removing code which is doing content comparison and use it as debug dump, that user and generate with example command:

berkeley_migration_verifier debug compare-ledgers...

File: src/app/berkeley_migration_verifier/berkeley_migration_verifier.ml [hunk 1]
```diff
@@ -403,7 +360,7 @@ let fork_block_height_exn ~fork_config_file =
   fork_config_exn ~fork_config_file |> member "blockchain_length" |> to_int
 
 let post_fork_validations ~mainnet_archive_uri ~migrated_archive_uri
-    ~migrated_replayer_output ~fork_config_file () =
+    ~migrated_replayer_output ~fork_config_file ~fork_ledger_hash () =
```
[the remark this answers is not in the document]
Comment (ghost-not-in-the-shell) [reply src/app/berkeley_migration_verifier/berkeley_migration_verifier.ml:363]: Do we have ledger hash in fork config?

**Comment by ghost-not-in-the-shell** [comment]
!ci-build-me

**Comment by ghost-not-in-the-shell** [comment]
!ci-build-me

**Review by joaosreis — APPROVED** [review]

File: src/app/berkeley_migration_verifier/berkeley_migration_verifier.ml (hunk 1 above)
[the remark this answers is not in the document]
Comment (dkijania) [reply src/app/berkeley_migration_verifier/berkeley_migration_verifier.ml:445]: oh no, we don't have, i confused it with state_hash. How we can provide it then based on fork config?

[the remark this answers is not in the document]
Comment (ghost-not-in-the-shell) [reply src/app/berkeley_migration_verifier/berkeley_migration_verifier.ml:445]: We cannot get this from the fork config, but we can get it from the genesis ledger config of the forked network. The fork config is the json file exported from the old network. I think we might want to use the genesis ledger config instead.

**Comment by ghost-not-in-the-shell** [comment]
!ci-build-me

**Comment by ghost-not-in-the-shell** [comment]
!ci-build-me

**Review by dkijania — APPROVED** [review]

**Comment by dkijania** [comment]
!ci-build-me

**Comment by ghost-not-in-the-shell** [comment]
!ci-build-me

**Comment by deepthiskumar** [comment]
!approved-for-mainnet

**Pull request merged**